pkg/clockcache/cache.go

```diff
@@ -131,6 +133,21 @@ func (self *Cache) insert(key interface{}, value interface{}, size uint64) (exis
 	return insertLocation, true, true
 }
 
+// RemoveUsingValue removes the entry in the cache by seeing the value part of the cache.
+// It goes through the entire cache and fills the value of each entry of the cache
+// in the given 'match()'. If the 'match()' returns true, the entry is marked as removed.
+func (self *Cache) RemoveUsingValue(match func(cacheValue interface{}) bool) {
+	self.elementsLock.Lock()
+	defer self.elementsLock.Unlock()
+	for i := range self.storage {
```

Comment:
Yikes. Not sure we want to take this path. We are fully locking the whole cache and iterating over probably millions of elements.

Reply:
This will result blocking all the ingest.